kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/cloud/CloudClient.java

```diff
@@ -124,6 +124,33 @@
     public int publish(String appTopic, KuraPayload payload, int qos, boolean retain) throws KuraException;
 
     /**
+     * Publishes a message to the remote server using the default priority 5.
+     * Before passing the message the to {@link org.eclipse.kura.data.DataService},
+     * the CloudClient will manipulate the provided topic by appending the necessary parts
+     * to achieve topic partitioning and device identification. It is also responsible to
+     * encode the {@link KuraPayload} payload into binary format.
+     * <br>
+     * The KuraStoreCapacityReachedException is thrown if the database buffer
+     * has reached its capacity for messages that are not yet published or
+     * they are still in transit.
+     *
+     * @param deviceId
+     *            A String specifying the device ID.
+     * @param appTopic
+     *            A String specifying the application portion of the topic the message is published on.
+     * @param payload
+     *            An KuraPayload representing the message to be published
+     * @param qos
+     *            An integer specifying the quality of service the message was published on.
+     * @param retain
+     *            Whether or not the broker should retain the message
+     * @return The published message's ID.
+     * @since {@link org.eclipse.kura.cloud} 1.1.0
+     */
+    public int publish(String deviceId, String appTopic, KuraPayload payload, int qos, boolean retain)
```

Comment:
It came out as a copy paste of the existing apis. Just adding the device Id. But I agree, we need to update the documentation with the description of the exception. I'll do that.